### DISCUSSION TITLE
How to access script objects statically?

### DISCUSSION THREAD

**Discussion by Zeon8** [ClearFoundry/ClearScript#247] [Q&A] [answered]
Hello! I used  `engine.Script` for accessing script objects.
I want to use ClearScript in environment where jitting isn't possible and how to access script objects statically?

**Comment by ClearScriptLib** [comment, marked as answer]
Hi @Zeon-4,

All script objects are instances of [`ScriptObject`](https://clearscript.clearfoundry.net/Reference/html/T_Microsoft_ClearScript_ScriptObject.htm). You can use that class as an alternative to `dynamic`:

```C#
var root = (ScriptObject)engine.Script;
var math = (ScriptObject)root.GetProperty("Math");
Console.WriteLine(math.GetProperty("PI"));
```

Good luck!

**Reply by Zeon8** [reply]
Thanks!
